thrust/examples/CMakeLists.txt

```diff
@@ -92,6 +92,11 @@ function(thrust_add_example target_name_var example_name example_src thrust_targ
     thrust_configure_cuda_target(${example_target} RDC ${THRUST_FORCE_RDC})
   endif()
 
+  # We do not want to explicitly include `host_device.h` if not needed, so force include the file for non CUDA targets
+  target_compile_options(${example_target} PRIVATE
```

Comment:
@alliepiper Is that fine? I believe we do not want to add all the noise of the `host_device` includes to all the examples

Reply:
I think it's good to show in the examples that users need to be mindful of `__host__ __device__` annotations when targeting multiple backends. I'd almost prefer directly including the contents of `host_device.h` in each example, without the `_CCCL` macros, to show users how to implement this themselves.

It's noisy and repetitive, but might be worth doing with a good comment explaining why/when it's needed.

Reply:
Then we should move the examples, what does `device_vector` even mean in the context of TBB and openMP?

Reply:
My point is that anything that uses a `device_vector` should live in `examples/cuda` and then use appropriately annotated functions.

But anything in `examples` should work out of the box for any backend without any magic intervention